sql/handler.cc

```diff
@@ -2936,6 +2936,12 @@ int handler::ha_rnd_next(uchar *buf)
 
   MYSQL_TABLE_IO_WAIT(m_psi, PSI_TABLE_FETCH_ROW, MAX_KEY, 0,
     { result= rnd_next(buf); })
+
+  if (!result)
```

Comment:
IIRC, at the InnoDB level UNIV_UNLIKELY has been added to the corresponding snippet after branch misprediction profiling, i.e. it's useful. Maybe here we should do `if (likely(!result))` as well